core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -516,19 +517,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
                 else if (config.interBrokerProtocolVersion < KAFKA_2_2_IV0) 1
                 else 2
 
-              val controlledShutdownRequest = new ControlledShutdownRequest.Builder(config.brokerId,
-                kafkaController.brokerEpoch, controlledShutdownApiVersion)
+              val controlledShutdownRequest = new ControlledShutdownRequest.Builder(
```

Comment:
we need to pass ```controlledShutdownApiVersion``` value to the Builder to retain previous behavior ?

Reply:
Thanks for the review. Yes you are right! I've pushed an update